src/main/java/org/spongepowered/api/item/recipe/RecipeRegistry.java

```diff
@@ -27,29 +27,32 @@
 import java.util.Set;
 
 /**
- * A RecipeRegistry holds all registered recipes for a given game.
+ * Used for {@link SmeltingRegistry} and {@link CraftingRegistry}
  */
 public interface RecipeRegistry {
 
     /**
      * Registers the given Recipe to make it available to craft.
      *
      * @param recipe The Recipe to register
+     * @throws IllegalArgumentException if it's the wrong
+     *     type of recipe for this registry
      */
-    void register(Recipe recipe);
+    void register(Recipe recipe) throws IllegalArgumentException;
```

Comment:
This is screaming about how it should use generics instead of just a raw `Recipe` type.